### PR TITLE
Correct Safari data for HTML element APIs

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR corrects the Safari data for the HTML element APIs based upon results from the mdn-bcd-collector project (using results from Safari 3-14), including mirroring to Safari iOS.

### CODE REVIEW

File: api/HTMLAreaElement.json [hunk 1]
```diff
@@ -412,7 +412,7 @@
               "version_added": "41"
             },
             "safari": {
-              "version_added": "11.1"
+              "version_added": false
```
Comment (foolip) [inline api/HTMLAreaElement.json:415]: Yep, no `referrerPolicy` in WebKit's HTMLAreaElement.idl.

File: api/HTMLAnchorElement.json [hunk 2]
```diff
@@ -412,10 +412,10 @@
               "version_added": "41"
             },
             "safari": {
-              "version_added": "11.1"
+              "version_added": "14"
```
Comment (foolip) [inline api/HTMLAnchorElement.json:415]: Added in https://trac.webkit.org/changeset/257707/webkit, 11.1 can't have been right.

File: api/HTMLButtonElement.json [hunk 3]
```diff
@@ -654,10 +654,10 @@
               "version_added": "27"
             },
             "safari": {
-              "version_added": "10"
+              "version_added": "10.1"
```
Comment (foolip) [inline api/HTMLButtonElement.json:657]: I don't trust these changes unless we've tested both Safari 10 and 10.1. Does the updater script set a ≤ 10.1 range here, or might we be setting the wrong version in lots of places without any indication that the underlying data is incomplete?

File: api/HTMLButtonElement.json [hunk 4]
```diff
@@ -942,10 +942,10 @@
               "version_added": "≤12.1"
             },
             "safari": {
-              "version_added": "3.2"
+              "version_added": "4"
```
Comment (foolip) [inline api/HTMLButtonElement.json:945]: How confident are you that this wasn't supported in Safari 3? Which platform was Safari 3 tested on?

https://trac.webkit.org/changeset/35381/webkit added the `willValidate` attribute to a bunch of interfaces, and the timing of it is consistent with is being in Safari 3.2 and Safari for iOS 3.

File: api/HTMLElement.json [hunk 5]
```diff
@@ -1969,10 +1969,10 @@
               "version_added": "45"
             },
             "safari": {
-              "version_added": "10"
+              "version_added": false
```
Comment (foolip) [inline api/HTMLElement.json:1972]: Where did the "10" come from. WebKit source has `nonce` in some IDL files, so this is a more complicated story that needs more digging.

File: api/HTMLFieldSetElement.json [hunk 6]
```diff
@@ -316,10 +316,10 @@
               "version_added": "27"
             },
             "safari": {
-              "version_added": "10"
+              "version_added": "10.1"
```
Comment (foolip) [inline api/HTMLFieldSetElement.json:319]: This is a minor version bump, did we test both?

File: api/HTMLIFrameElement.json [hunk 7]
```diff
@@ -184,10 +184,10 @@
               "version_added": "25"
             },
             "safari": {
-              "version_added": "10"
+              "version_added": "10.1"
```
Comment (foolip) [inline api/HTMLIFrameElement.json:187]: This is a minor version bump, did we test both?

File: api/HTMLInputElement.json [hunk 8]
```diff
@@ -943,10 +943,10 @@
               "version_added": "27"
             },
             "safari": {
-              "version_added": "10"
+              "version_added": "10.1"
```
Comment (foolip) [inline api/HTMLInputElement.json:946]: This is a minor version bump, did we test both?

File: api/HTMLElement.json [hunk 9]
```diff
@@ -599,10 +599,10 @@
               "version_added": true
             },
             "safari": {
-              "version_added": "6"
+              "version_added": "3"
```
Comment (foolip) [inline api/HTMLElement.json:602]: Since Safari 3 was the earliest version tested for this PR, can we really say this? Let's use &le;4 where it's not clear what the status of Safari 1-2 were.

File: api/HTMLMeterElement.json [hunk 10]
```diff
@@ -26,7 +26,10 @@
             "version_added": "11"
           },
           "safari": {
-            "version_added": "5"
+            "version_added": "6"
```
Comment (foolip) [inline api/HTMLMeterElement.json:29]: Can you test if `<meter>` itself was also added in Safari 6 and if that entry needs updating? (Maybe split into a separate PR if it's complicated.)

File: api/HTMLScriptElement.json [hunk 11]
```diff
@@ -169,10 +169,10 @@
               "version_added": "14"
             },
             "safari": {
-              "version_added": "4"
+              "version_added": "6"
```
Comment (foolip) [inline api/HTMLScriptElement.json:172]: Can you check where the original "4" came from? Perhaps the content attribute was supported earlier?

**Review by foolip — COMMENTED** [review]
So far I've looked only at the cases where the claimed support is being shrunk, since false negatives from our tests are much more likely than false positives.

A bunch of comments, but I stopped asking about 10 vs 10.x after a while so not all of those are covered.

This is fairly big, so I'd suggest splitting off anything that cannot be easily resolved, to leave us only with simple cases.

**Comment by foolip** [comment]
It would be great to separate out a "supported since earliest tested version of Safari" subset of this PR, since that's a lot of it and easy to review.

File: api/HTMLElement.json (hunk 9 above)
Comment (vinyldarkscratch) [reply api/HTMLElement.json:602]: The `HTMLElement` API wasn't supported until Safari 3, so "1-2" is not possible -- I wouldn't set it so confidently otherwise.

File: api/HTMLButtonElement.json (hunk 3 above)
Comment (vinyldarkscratch) [reply api/HTMLButtonElement.json:657]: The collector tests both 10 and 10.1; otherwise, it would set a range to ≤10.1, which would cause no modifications to the data.

File: api/HTMLButtonElement.json (hunk 4 above)
Comment (vinyldarkscratch) [reply api/HTMLButtonElement.json:945]: I tested these in Mac OS X on Safari 3.2.3 (which is the latest 3.2 release).

File: api/HTMLButtonElement.json (hunk 3 above)
Comment (foolip) [reply api/HTMLButtonElement.json:657]: Sweet!

File: api/HTMLButtonElement.json (hunk 4 above)
Comment (foolip) [reply api/HTMLButtonElement.json:945]: Continued in https://github.com/mdn/browser-compat-data/pull/.
